### PR TITLE
Add support for displaying custom order status labels

### LINKED ISSUE #87: Custom order statuses
minimally support the display of custom status labels but not support any custom workflows

v3 API does support localized labels

### PR DESCRIPTION
**Pull request by AmandaRiu**
Fixes #87 and adds support for displaying custom order status labels. The hard-coded core order status labels have been replaced by a FluxC action to fetch and store order statuses and their appropriate label from the api. 

~~**NOTE: The [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1101) must be merged before this can be merged**~~

What to expect: The raw order status will temporarily be displayed the very first time the data is fetched from the api, but will refresh almost immediately. Same goes for new custom order statuses. 

Update release notes:
- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

## How it works
The db is queried for the `WCOrderStatusModel` that matches the `status` property of an order. If none exists, the raw `status` will be displayed and a request will be fired to fetch order statuses from the api. When the `OnOrderStatusChanged` event is received, the labels will be refreshed with the values from the db.

Requests to fetch a fresh batch of order status options from the api happen:
- On app launch.
- When an order status is encountered that is not already saved in the database.
- The site is changed (since fluxC only fetches for the active site)

Other cool stuff included:
- A new internal event was created to listen for a site change - `SelectedSiteChangedEvent `. This is fired anytime `SelectedSite.set()` is called.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/WooCommerce.kt [hunk 1]
```diff
@@ -125,6 +127,11 @@ open class WooCommerce : MultiDexApplication(), HasActivityInjector, HasServiceI
                 this, BuildConfig.ZENDESK_DOMAIN, BuildConfig.ZENDESK_APP_ID,
                 BuildConfig.ZENDESK_OAUTH_CLIENT_ID
         )
+
+        site?.let { // Fetch order status options on app launch
```
Comment (nbradbury) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/WooCommerce.kt]: I'm wondering if this belongs in `onFirstActivityResumed()`?

**Comment by nbradbury** [comment]
@AmandaRiu I'm finished with my first round. Looks good, but should `OrderStatusFilterDialog` be updated as well? Right now it's using the hard-coded order statuses.

Also, there are now conflicts that need to be resolved.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/WooCommerce.kt (hunk 1 above)
Comment (AmandaRiu) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/WooCommerce.kt]: I agree. Moved in 243b4dfab5fa22911a603aec69a16de2b3d41004

**Comment by AmandaRiu** [comment]
@nbradbury Good call on the filter dialog. A nice side effect of this is now users can filter by custom order status as well! Ready for the next round.

**Comment by nbradbury** [comment]
Looks good - nice work! :shipit:

**Pull request merged**